`$(SkipTests)` isn't respected if `UseVSTestRunner=true`

<!-- If these statements apply, replace [ ] with [x] before filing your issue. -->
- [x] This issue is blocking <!-- Describe below what is blocked. -->
- [ ] This issue is causing unreasonable pain

<!-- Write your issue description below. -->

If `<UseVSTestRunner>true</UseVSTestRunner>` is set, VSTest.targets gets imported:
https://github.com/dotnet/arcade/blob/b31c4466a49800c2f019690e81182c7ffd584dda/src/Microsoft.DotNet.Arcade.Sdk/tools/XUnit/XUnit.targets#L38

If a test project is marked with `SkipTests`, a test will fail due to an invalid command constructed by VSTest.targets:
```
C:\Users\user\.nuget\packages\microsoft.dotnet.arcade.sdk\8.0.0-beta.23067.5\tools\VSTest.targets(46,5): error MSB3073: The command "echo === COMMAND LINE === >  [D:\dev\test\Console.Tests.csproj]
C:\Users\user\.nuget\packages\microsoft.dotnet.arcade.sdk\8.0.0-beta.23067.5\tools\VSTest.targets(46,5): error MSB3073:                    echo "D:\dev\test\.dotnet\dotnet.exe" 
 test  --logger:"console;verbosity=normal" --logger:"trx;LogFileName=" --logger:"html;LogFileName=" "--ResultsDirectory:" "--Framework:,Version=" >> "" 2>&1 >> " exited with code 255. [D:\dev\test\Console.Tests.csproj]
```

Here's an [example](https://dev.azure.com/dnceng/internal/_build/results?buildId=2106375&view=logs&j=d6d0747d-862c-5953-c201-f4e1a93f2cef&t=218889c2-d554-53f8-5a01-b0738d33beaa).


The fix is to condition [`RunTests` target](https://github.com/dotnet/arcade/blob/b31c4466a49800c2f019690e81182c7ffd584dda/src/Microsoft.DotNet.Arcade.Sdk/tools/VSTest.targets#L3-L4) as below:
```diff
  <Target Name="RunTests"
          Outputs="%(TestToRun.ResultsStdOutPath)"
+         Condition="'$(SkipTests)' != 'true'">
    <PropertyGroup>
```


<!-- DO NOT DELETE -->
<!-- For internal use only; put release notes here. -->
<!-- For guidance on writing good release notes, please see documentation here: https://dev.azure.com/dnceng/internal/_wiki/wikis/DNCEng%20Services%20Wiki/983/ReleaseNotesGuidance -->
<!-- Additionally, please specify the note category below. -->
### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [ ] Internal Infrastructure Improvements
### Release Note Description

--- a/src/Microsoft.DotNet.Arcade.Sdk/tools/VSTest.targets
+++ b/src/Microsoft.DotNet.Arcade.Sdk/tools/VSTest.targets
@@ -1,7 +1,8 @@
 <!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. -->
 <Project>
   <Target Name="RunTests"
-          Outputs="%(TestToRun.ResultsStdOutPath)">
+          Outputs="%(TestToRun.ResultsStdOutPath)"
+          Condition="'$(SkipTests)' != 'true'">
     <PropertyGroup>
       <_TestEnvironment>%(TestToRun.EnvironmentDisplay)</_TestEnvironment>
       <_TestAssembly>%(TestToRun.Identity)</_TestAssembly>

